plugins/junit/src/com/intellij/execution/junit/codeInsight/JUnit5MalformedOrderedTestInspection.kt

```diff
@@ -0,0 +1,102 @@
+// Copyright 2000-2017 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.intellij.execution.junit.codeInsight
+
+import com.intellij.codeInsight.AnnotationUtil
+import com.intellij.codeInspection.AbstractBaseJavaLocalInspectionTool
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.*
+import com.intellij.psi.codeStyle.CodeStyleManager
+import com.intellij.psi.codeStyle.JavaCodeStyleManager
+import com.intellij.psi.impl.PsiClassImplUtil
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.PsiTypesUtil
+import com.intellij.psi.util.PsiUtil
+import com.siyeh.ig.junit.JUnitCommonClassNames
+
+class JUnit5MalformedOrderedTestInspection : AbstractBaseJavaLocalInspectionTool() {
+  override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+    val file = holder.file
+    if (JavaPsiFacade.getInstance(file.project).findClass(JUnitCommonClassNames.ORG_JUNIT_JUPITER_API_TEST, file.resolveScope) == null) {
+      return PsiElementVisitor.EMPTY_VISITOR
+    }
+    val psiClassOrderAnnotationOrderer = JavaPsiFacade.getInstance(file.project)
+                                           .findClass("org.junit.jupiter.api.MethodOrderer.OrderAnnotation", file.resolveScope)
+                                         ?: return PsiElementVisitor.EMPTY_VISITOR
+    return object : JavaElementVisitor() {
+      override fun visitMethod(method: PsiMethod) {
+
+        AnnotationUtil.findAnnotation(method, JUnitCommonClassNames.ORG_JUNIT_JUPITER_API_TEST) ?: return
+        val orderAnnotation = AnnotationUtil.findAnnotation(method, "org.junit.jupiter.api.Order") ?: return
+        val psiClass = method.containingClass ?: return
+
+        val testMethodOrderAnnotation = AnnotationUtil.findAnnotation(psiClass, "org.junit.jupiter.api.TestMethodOrder")
+        if (testMethodOrderAnnotation == null) {
```

Comment:
from the javadoc of `TestMethodOrder`: one can put annotation to the superclass/superinterface, it's not covered inside `AnnotationUtil`

Reply:
sorry, I have missed JavaDoc ... it also seems that I have to support `@RepeatedTest`, `@ParameterizedTest`, `@TestFactory`, and `@TestTemplate`

Reply:
I also added checks in inheritance hierarchy and more tests (seems that all combinations of inheritance vs annotation and type of testable class is kind of rabbit hole).